### PR TITLE
Font scaling and reimplemented orb colors

### PR DESCRIPTION
**Pull request by CelticMinstrel**
This implements a new preference which scales all fonts in the game up by a chosen factor (up to 200%; 300% caused the title screen to crash from lack of space), and reimplements the orb colour selection options as a new GUI2 dialog. I also added a minor feature to the friends list pane - clicking an entry copies it into the text box for easy editing.

The orb colour selection needs someone else to verify that it works correctly, since recolouring doesn't work on my build for some reason. I also want to make some changes to the layout.

There's still a minor bug with the font scaling preference as well. I already know what causes it and how to fix it, so I'll push that in the next day or two.

### CODE REVIEW

**Comment by CelticMinstrel** [comment]
Fixed the scaling bug and the orbs layout, so now all that's left is to wait for someone to test it; then I can squash the fixups and merge this.

File: src/gui/dialogs/select_orb_colors.cpp [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+void tselect_orb_colors::pre_show(CVideo&, twindow& window)
+{
+	setup_orb_group("unmoved", show_unmoved_, unmoved_, window);
+	setup_orb_group("partial", show_partial_, partial_, window);
+	setup_orb_group("moved", show_moved_, moved_, window);
+	setup_orb_group("ally", show_ally_, ally_, window);
+	setup_orb_group("enemy", show_enemy_, enemy_, window);
+	
+	tbutton& reset = find_widget<tbutton>(&window, "orb_defaults", false);
+	event::connect_signal_mouse_left_click(reset, boost::bind(
```
Comment (Vultraz) [inline src/gui/dialogs/select_orb_colors.cpp]: You don't need to have reset as a separate variable here. You can do this:

`connect_signal_mouse_left_click(find_widget<tbutton>(&window, "orb_defaults", false),
		boost::bind(&tselect_orb_colors::handle_reset_click,
 		this, boost::ref(window)))`

Notice the lack of the class specifier too.

File: src/gui/dialogs/select_orb_colors.cpp [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+		twidget* next = iter->get(twalker_::child);
+		if(ttoggle_button* button = dynamic_cast<ttoggle_button*>(next)) {
+			group.push_back(button);
+			if(button->id().rfind("_" + color) != std::string::npos) {
+				button->set_value_bool(true);
+			} else {
+				button->set_value_bool(false);
+			}
+			if(connect) {
+				event::connect_signal_mouse_left_click(*button, boost::bind(
```
Comment (Vultraz) [inline src/gui/dialogs/select_orb_colors.cpp]: Shouldn't have a class specifier

**Comment by Vultraz** [comment]
Please add this commit to this PR: https://github.com/Vultraz/wesnoth/commit/80ad4a2407961bdb7089496cda61c2dfde2a6e79

It significantly improves the look of the orb buttons.

File: src/gui/dialogs/select_orb_colors.cpp [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+	event::connect_signal_mouse_left_click(reset, boost::bind(
+		&tselect_orb_colors::handle_reset_click,
+		this, boost::ref(window)
+	));
+}
+
+void tselect_orb_colors::display(CVideo& video)
+{
+	tselect_orb_colors dialog;
+	if(dialog.show(video)) {
```
Comment (Vultraz) [inline src/gui/dialogs/select_orb_colors.cpp]: As mentioned on IRC, what's this line for?

File: src/gui/dialogs/preferences_dialog.cpp [hunk 4]
```diff
@@ -638,6 +655,12 @@ void tpreferences::initialize_members(twindow& window)
 			boost::ref(textbox),
 			boost::ref(window)));
 
+	friend_list.set_callback_value_change(boost::bind(
```
Comment (Vultraz) [inline src/gui/dialogs/preferences_dialog.cpp]: Inconsistent coding style.

**Comment by Vultraz** [comment]
Bug: it's now impossible to add any friend to the friends list; you get the Invalid Username error each time.

**Comment by Vultraz** [comment]
And finally, as mentioned on IRC, 'Defaults' does not save the default orb state and the previous state will be displayed when opening the dialog the next time.

File: src/gui/dialogs/select_orb_colors.cpp (hunk 3 above)
Comment (CelticMinstrel) [inline src/gui/dialogs/select_orb_colors.cpp]: This block saves the orbs to the preferences. If you clicked Cancel, it doesn't run.

File: src/gui/dialogs/select_orb_colors.cpp (hunk 1 above)
Comment (CelticMinstrel) [inline src/gui/dialogs/select_orb_colors.cpp]: Yes, there are infinite varieties in which to code a thing. This way works perfectly fine.

I had a reason for the `event::` bit but I can't remember what it was now. It's not hurting anyone to leave it.

File: src/gui/dialogs/select_orb_colors.cpp (hunk 2 above)
Comment (CelticMinstrel) [inline src/gui/dialogs/select_orb_colors.cpp]: It's not hurting anyone. I think there was a reason or something for it to be there.

File: src/gui/dialogs/preferences_dialog.cpp (hunk 4 above)
Comment (CelticMinstrel) [inline src/gui/dialogs/preferences_dialog.cpp]: What are you talking about?

**Comment by CelticMinstrel** [comment]
> Also of note: while clicking on a friends list entry does make the name appear in the box, if you make changes and select 'Add Friend', a new entry is created.

This is presumably only if you make changes to the first word, right? I'm not sure if it's worth fixing this right now.

File: src/gui/dialogs/preferences_dialog.cpp (hunk 4 above)
Comment (Vultraz) [inline src/gui/dialogs/preferences_dialog.cpp]: IMO that should be handled in the post_show function. The display function isn't meant to be implemented outside the header.

File: src/gui/dialogs/select_orb_colors.cpp (hunk 3 above)
Comment (Vultraz) [inline src/gui/dialogs/select_orb_colors.cpp]: Such a thing should be handled in the post_show function. The display() function isn't supposed to be implemented outside the header.

**Comment by Vultraz** [comment]
> This is presumably only if you make changes to the first word, right? I'm not sure if it's worth fixing this right now.

Yes, that's right.

**Comment by CelticMinstrel** [comment]
Oh, also:

> And finally, as mentioned on IRC, 'Defaults' does not save the default orb state and the previous state will be displayed when opening the dialog the next time.

This is when you click Defaults and then OK?

File: src/gui/dialogs/select_orb_colors.cpp (hunk 1 above)
Comment (Vultraz) [inline src/gui/dialogs/select_orb_colors.cpp]: I'd prefer we try to keep a consistent coding style throughout GUI2 dialogs for easy future conversion.

File: src/gui/dialogs/select_orb_colors.cpp (hunk 3 above)
Comment (CelticMinstrel) [inline src/gui/dialogs/select_orb_colors.cpp]: I couldn't figure out any way to get the dialog result in the post_show function.

File: src/gui/dialogs/preferences_dialog.cpp (hunk 4 above)
Comment (Vultraz) [inline src/gui/dialogs/preferences_dialog.cpp]: My mistake, ignore this.

**Comment by Vultraz** [comment]
Let me check.

File: src/gui/dialogs/select_orb_colors.cpp (hunk 3 above)
Comment (Vultraz) [inline src/gui/dialogs/select_orb_colors.cpp]: I believe you need `get_retval() != twindow::OK`

**Comment by Vultraz** [comment]
I was mistaken, it seems. I must've been clicking cancel.

File: data/gui/default/window/preferences/02_display.cfg [hunk 5]
```diff
@@ -208,6 +208,20 @@
 			[/grid]
 		[/column]
 	[/row]
+
```
Comment (Vultraz) [inline data/gui/default/window/preferences/02_display.cfg]: Could you add a spacer row (`_GUI_PREFERENCES_SPACER_ROW`) here?

**Comment by CelticMinstrel** [comment]
I don't know why the latest Travis build failed, but it doesn't seem to be anything about the latest commit (as far as I can tell, those errors imply the failure should've also occurred on the previous build), so hopefully it's something transient.

**Pull request merged**